csv/shared/src/main/scala/fs2/data/csv/package.scala

```diff
@@ -217,6 +221,27 @@ package object csv {
     }
   }
 
+  /** Encode a specified type into a CSV prepending the given headers. */
+  def encodeWithGivenHeaders[T]: PartiallyAppliedEncodeWithGivenHeaders[T] =
```

Comment:
Wdyt about renaming to ``encodeWithHeaders`? We already have `encodeWithoutHeaders`, so the naming feels a bit more consistent then the `WithGiven`.

No strong opinion on this though, just noticed it, decide as you see fit.

Reply:
I wanted to keep consistent naming between the high and low level naming. Since the low level naming cannot be `writeWithHeaders`(this is the deprecated one) and high-level cannot be `encodeGivenHeaders` (this is the deprecated one), I settled for merging both.